### PR TITLE
deploy phalanx to usdf-cm and usdf-cm-dev vclusters.

### PR DESCRIPTION
**Pull request by pav511**
adjust helm command to not timeout if charts exist.

### CODE REVIEW

File: applications/argocd/values-usdf-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        # Optional set of OIDC claims to request on the ID token.
+        requestedIDTokenClaims: {"groups": {"essential": true}}
+    rbac:
+      policy.csv: |
+        g, ytl@slac.stanford.edu, role:admin
+        g, ppascual@slac.stanford.edu, role:admin
+        g, pav@slac.stanford.edu, role:admin
+        g, reinking@slac.stanford.edu, role:admin
+        g, fritzm@slac.stanford.edu, role:admin
+        g, reinking@slac.stanford.edu, role:admin
```
Comment (fritzm) [inline applications/argocd/values-usdf-cm.yaml:30]: Heinrich is listed twice here?

Comment (pav511) [reply applications/argocd/values-usdf-cm.yaml:30]: good catch I'll delete

**Comment by pav511** [comment]
ah these are the tests that fail from the addition made to the helm command

File: applications/vault-secrets-operator/values-usdf-cm-dev.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    - name: "VAULT_ROLE_ID"
+      valueFrom:
+        secretKeyRef:
+          name: "vault-credentials"
+          key: "VAULT_ROLE_ID"
+    - name: "VAULT_SECRET_ID"
+      valueFrom:
+        secretKeyRef:
+          name: "vault-credentials"
+          key: "VAULT_SECRET_ID"
```
Comment (rra) [inline applications/vault-secrets-operator/values-usdf-cm-dev.yaml:12]: This is the default, so you can delete all of this and just have an empty file.

File: applications/vault-secrets-operator/values-usdf-cm.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    - name: "VAULT_ROLE_ID"
+      valueFrom:
+        secretKeyRef:
+          name: "vault-credentials"
+          key: "VAULT_ROLE_ID"
+    - name: "VAULT_SECRET_ID"
+      valueFrom:
+        secretKeyRef:
+          name: "vault-credentials"
+          key: "VAULT_SECRET_ID"
```
Comment (rra) [inline applications/vault-secrets-operator/values-usdf-cm.yaml:12]: Likewise here, you can delete all of this.

File: environments/values-usdf-cm-dev.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+applications:
+  # This environment uses an ingress managed in a separate Kubernetes cluster,
+  # despite that configuration not being officially supported by Phalanx.
+  cert-manager: false
+  gafaelfawr: false
+  ingress-nginx: false
+
+  argocd: true
+  vault-secrets-operator: true
```
Comment (rra) [inline environments/values-usdf-cm-dev.yaml:15]: These are mandatory applications so you shouldn't enable them here. They're always enabled.

File: environments/values-usdf-cm.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+applications:
+  # This environment uses an ingress managed in a separate Kubernetes cluster,
+  # despite that configuration not being officially supported by Phalanx.
+  cert-manager: false
+  gafaelfawr: false
+  ingress-nginx: false
+
+  argocd: true
+  vault-secrets-operator: true
```
Comment (rra) [inline environments/values-usdf-cm.yaml:15]: Likewise here, delete that part.

File: environments/values-usdf-cm-dev.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+name: "usdf-cm-dev"
+appOfAppsName: "science-platform"
+vaultUrl: "https://vault.slac.stanford.edu"
+vaultPathPrefix: "secret/rubin/usdf-cm-dev"
+
+applications:
+  # This environment uses an ingress managed in a separate Kubernetes cluster,
+  # despite that configuration not being officially supported by Phalanx.
+  cert-manager: false
+  gafaelfawr: false
```
Comment (rra) [inline environments/values-usdf-cm-dev.yaml:11]: I think Fritz said they do need Gafaelfawr?

**Review by rra — APPROVED** [review]
Mostly looks good, just some simplifications to the vault-secrets-operator and environments configuration to do before merging. Also, I think they do want Gafaelfawr in those environments? But that can be done as a separate PR.

**Pull request merged**